tests/test_bdist_pex.py

```diff
@@ -81,3 +83,52 @@ def test_pex_args_shebang_with_spaces():
 
 def test_pex_args_shebang_without_spaces():
   assert_pex_args_shebang('#!/usr/bin/python')
+
+
+def test_unwriteable_contents():
+  my_app_setup_py = dedent("""
+      from setuptools import setup
+
+      setup(
+        name='my_app',
+        version='0.0.0',
+        zip_safe=True,
+        packages=['my_app'],
+        include_package_data=True,
+        package_data={'my_app': ['unwriteable.so']},
+      )
+    """)
+
+  UNWRITEABLE_PERMS = 0o400
+  with temporary_content({'setup.py': my_app_setup_py,
+                          'my_app/__init__.py': '',
+                          'my_app/unwriteable.so': 42},
```

Comment:
What does 42 mean? Consider defining as a constant to avoid magic numbers.

Reply:
It's not magic - it's random:
https://github.com/pantsbuild/pex/blob/cbf690845d40da54b3c7a3c2704c55a4a9dc614b/pex/testing.py#L68-L80
I think FORTY_TWO_BYTES_OF_RANDOM_DATA is perhaps over the top in this case.

Reply:
Oh goodness we not trying to write Java 😂 But something shorter like `random_data` I think would be helpful.

Reply:
Hrm - thats just as bad almost.
```
random_data = 42
with temporary_content({'setup.py': my_app_setup_py,
                          'my_app/__init__.py': '',
                          'my_app/unwriteable.so': random_data},
...
```
This is a lie. We are not writing `42` as the random data in `my_app/unwriteable.so`, we're writing 42 bytes of random data.

I'll switch to ''. But note!: it's just as magic and yet most folks don't balk at it. There is something to be said for familiarity with the testing libs in use (`temporary_content` in this case). You can't spell _everything_ out. This is coming from someone who likes explicit more than most! All this said you are in the position of not knowing this codebase and being kind enough to review, so I completely understand the comments.